File: Numbers/GCD.cpp
#include <bits/stdc++.h>
using namespace std;

/*
int findGCD(int a, int b) {
    for (int i = min(a, b); i > 0; i--) {
        if (a % i == 0 && b % i == 0) {
            return i;
            break;
        }
    }
    // GCD of any number with 0 is the number itself
    return 1; 
}
*/

int findGCD(int a, int b){
    while(a>0 && b>0){
        if(a>b){
            a = a % b;
        }
        else{
            b = b % a;
        }
    }
    if(a==0) return b;
    return a;
}

int main() {
    int a, b;
    cin >> a >> b;

    int ans = findGCD(a, b);
    cout << ans << endl;

    return 0;
}

File: Java/Numbers/GCD.java
public class GCD {

    public static int better(int n1, int n2) {
        int gcd = 1;
        for (int i = Math.min(n2, n1); i > 1; i--) {
            if (n1 % i == 0 && n2 % i == 0) {
                gcd = i;
            }
        }

        return gcd;
    }

    public static int bruteForce(int n1, int n2) {
        int gcd = 1;

        for (int i = 1; i < Math.min(n1, n2); i++) {
            if (n1 % i == 0 && n2 % i == 0) {
                gcd = i;
            }
        }

        return gcd;
    }

    public static void main(String[] args) {
        int n1 = 9;
        int n2 = 12;
        System.out.println(better(n1, n2));
    }
}
